packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -137,15 +175,68 @@ export class DetachedFieldIndex {
 		return key;
 	}
 
+	/**
+	 * Returns all entries last created or used by the given revision.
+	 */
+	public getLatestRelevantRoots(revision: RevisionTag): [ForestRootId, Delta.DetachedNodeId][] {
+		return Array.from(this.latestRelevantRevisionToFields.get(revision)?.entries() ?? []);
+	}
+
+	/**
+	 * Returns all entries created by the given revision.
+	 */
+	public getRoots(revision: RevisionTag): ForestRootId[] {
+		return Array.from(this.detachedNodeToField.get(revision)?.values() ?? []).map(
+			({ root }) => root,
+		);
+	}
+
+	/**
+	 * Removes all entries created by the given revision, no matter what their latest
+	 * relevant revision is.
+	 */
+	public deleteRoots(revision: RevisionTag): void {
+		const entries = this.detachedNodeToField.get(revision);
+		assert(entries !== undefined, "no entries created by the given revision");
+		for (const [_, { root, latestRelevantRevision }] of entries.entries()) {
+			this.deleteRootFromLatestRelevantRevisionsMap(root, latestRelevantRevision);
+		}
+		this.detachedNodeToField.delete(revision);
+	}
+
 	public deleteEntry(nodeId: Delta.DetachedNodeId): void {
-		const found = deleteFromNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor);
-		assert(found, 0x7ab /* Unable to delete unknown entry */);
+		const errorMessage = "Unable to delete unknown entry";
```

Comment:
I'm afraid this will circumvent our assert message minification tooling.
I don't think you need to assert both on the tryGet and one the delete here, just the first one will do.